docs/design.md

```diff
@@ -0,0 +1,66 @@
+   
+   higher fee per transaction comes first, if transaction fee is the same, lower transaction number comes first, in that order, fill the cap and ignore the rest.
+   
+   By picking the transactions with higher transaction fee, it encourages batching, while allowing small transactions to have the opportunity to also be included if they are willing to pay a slightly higher transaction fee. Alternatives to this approach are highest fee per operation and first comes first serve, but they don't encourage batching and discourage writing near the end of a transaction time.
+
+#### Maximum batch size
+   
+   By defining a maximum number of operations per batch, the strategy circumvents participants to anchor arbitrarily large trees on the system. At its core, this mitigation strategy forces the attacker to deal with the organic economic pressure exerted by the underlying chain's transactional unit cost. Each instantiation of a Sidetree-based DID Method may select a different maximum batch size; the size for the default configuration is TBD. 
+
+#### Proof of Fee
+
+   Each Sidetree transaction on the target chain is required to include a deterministic fee, based on the number of DID operations they seek to include via the on-chain transaction.
+
+#### One Operation per DID per Batch
+  Only one operation per DID per batch is allowed, this prevents the operation chain of any DID from growing at an intractable rate.
+
+#### Commitment and Reveal for Operations
+  Upon DID creation, the create operation payload must include:
+  1. The hash of a _commitment_ value for the next recover operation.
+  1. The hash of a _commitment_ value for the next update operation.
+  https://en.wikipedia.org/wiki/Commitment_scheme
+
+  The DID owner must reproduce and reveal the correct commitment value in the subsequent operation for the operation to be considered valid. In addition, each subsequent operation must also include the hash of the new commitment value(s) for the next operation. This scheme enables efficient dismissal of counterfeit operations without needing to evaluate signatures.
+
+## Sidetree Client Guidelines
+A Sidetree client manages the private keys and performs document operations on behalf of the DID owner. The Sidetree client needs to comply to the following guidelines to keep the DIDs it manages secure.
+
+1. The client MUST keep the operation payload once it is submitted to a Sidetree node until it is generally available and observed. If the submitted operation is not observed, the same operation payload MUST be resubmitted. Submitting a different operation payload would put the DID in risk of a _late publish_ attack which can lead to an unrecoverable DID if the original operation payload contains a recovery key rotation and the recovery key is lost.
```

Comment:
Is this something that should go in the spec rather than the design docs of the reference implementation?

Reply:
@kdenhartog, yes I've been wanting to deprecate this section and only reason I haven't is because the content has yet to be moved/replicated across to the main spec. Let me file a doc bug explicitly to track this.

Reply:
Issue #975 filed.